pkgs/applications/graphics/tauray/default.nix

```diff
@@ -0,0 +1,65 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, graphviz
+, doxygen
+, pkg-config
+, python3
+, libcbor
+, czmq
+, glm
+, nng
+, mbedtls
+, openxr-loader
+, SDL2
+, shaderc
+, spirv-headers
+, vulkan-headers
+, vulkan-loader
+, wayland
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "tauray";
+  version = "1.0.1";
+
+  src = fetchFromGitHub {
+    owner = "vga-group";
+    repo = "tauray";
+    rev = "v${finalAttrs.version}";
+    sha256 = "sha256-Q3Dv96iptDvmBp9mIVSYzb7EnIzD1kFejwrFuc1HqdU=";
+    fetchSubmodules = true;
```

Comment:
How would I go about fetching only the ones that aren't in nixpkgs? Some of them also break compatibility frequently so they need to be pinned to specific versions according to upstream.

Reply:
The fetcher has no support for that. If it is only one it can be manually done. If there are multiple that is cumbersome. Are there options to build with a system variant? If not we could rm the replaced submodules in postFetch and then symlink ours into place.

Reply:
Unfortunately there are no options for that. I'd have to PR such a change upstream first.

IIRC when I asked about it the response I got was that vulkan-headers need to be pinned since tauray uses the C++ bindings which have changed multiple times without warning and glslang is used through its library interface which is unstable. Spirv-headers and OpenXR-SDK are probably okay to replace with a packaged version. `fuchsia_radix_sort` would need to be packaged manually but there are no changes on the horizon to it so it should be quite safe too.